# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/07 11:16:13-08:00 rddunlap@osdl.org 
#   [PATCH] i2c-ali1563: fix init & exit section usage
#   
#   Fix init & exit section usages, beginning with this diagnostic
#   from reference_discarded.pl (make buildcheck):
#   Error: ./drivers/i2c/busses/i2c-ali1563.o .data refers to 00000278 R_386_32          .exit.text
#   
#   Signed-off-by: Randy Dunlap <rddunlap@osdl.org>
#   Signed-off-by: Greg Kroah-Hartman <greg@kroah.com>
# 
# drivers/i2c/busses/i2c-ali1563.c
#   2005/01/06 21:42:07-08:00 rddunlap@osdl.org +4 -4
#   i2c-ali1563: fix init & exit section usage
# 
diff -Nru a/drivers/i2c/busses/i2c-ali1563.c b/drivers/i2c/busses/i2c-ali1563.c
--- a/drivers/i2c/busses/i2c-ali1563.c	2005-01-22 23:45:29 -08:00
+++ b/drivers/i2c/busses/i2c-ali1563.c	2005-01-22 23:45:29 -08:00
@@ -306,7 +306,7 @@
 	pci_write_config_word(dev,ALI1563_SMBBA,ctrl);
 }
 
-static int __init ali1563_setup(struct pci_dev * dev)
+static int __devinit ali1563_setup(struct pci_dev * dev)
 {
 	u16 ctrl;
 
@@ -362,7 +362,7 @@
 	.algo	= &ali1563_algorithm,
 };
 
-static int __init ali1563_probe(struct pci_dev * dev,
+static int __devinit ali1563_probe(struct pci_dev * dev,
 				const struct pci_device_id * id_table)
 {
 	int error;
@@ -378,7 +378,7 @@
 	return error;
 }
 
-static void __exit ali1563_remove(struct pci_dev * dev)
+static void __devexit ali1563_remove(struct pci_dev * dev)
 {
 	i2c_del_adapter(&ali1563_adapter);
 	ali1563_shutdown(dev);
@@ -395,7 +395,7 @@
  	.name		= "ali1563_i2c",
 	.id_table	= ali1563_id_table,
  	.probe		= ali1563_probe,
-	.remove		= ali1563_remove,
+	.remove		= __devexit_p(ali1563_remove),
 };
 
 static int __init ali1563_init(void)
